src/controllers/handlers/livekit-webhook-handler.ts

```diff
@@ -1,87 +1,73 @@
 import { DecentralandSignatureContext } from '@dcl/platform-crypto-middleware'
-import { HandlerContextWithPath } from '../../types'
-import { WebhookReceiver } from 'livekit-server-sdk'
+import { HandlerContextWithPath, IPeersRegistry } from '../../types'
 import { IHttpServerComponent } from '@well-known-components/interfaces'
-
-enum WebhookEvent {
-  ParticipantJoined = 'participant_joined',
-  ParticipantLeft = 'participant_left'
-}
+import { InvalidRequestError } from '@dcl/platform-server-commons'
+import { ParticipantEvent, WebhookEventName } from '../../adapters/livekit-client'
 
 const TOPIC_SUFFIX_BY_EVENT = {
-  [WebhookEvent.ParticipantJoined]: 'join',
-  [WebhookEvent.ParticipantLeft]: 'leave'
+  [WebhookEventName.ParticipantJoined]: 'join',
+  [WebhookEventName.ParticipantLeft]: 'leave'
 }
 
-function isValidEvent(event: string): event is keyof typeof WebhookEvent {
-  return Object.values(WebhookEvent).includes(event as WebhookEvent)
+function isValidEvent(event: string): event is ParticipantEvent {
+  return Object.values(WebhookEventName).includes(event as WebhookEventName)
 }
 
 export async function livekitWebhookHandler(
-  ctx: HandlerContextWithPath<'nats' | 'config' | 'logs', '/livekit-webhook'> & DecentralandSignatureContext<any>
+  ctx: HandlerContextWithPath<'nats' | 'logs' | 'livekitClient' | 'peersRegistry', '/livekit-webhook'> &
+    DecentralandSignatureContext<any>
 ): Promise<IHttpServerComponent.IResponse> {
   const {
-    components: { nats, config, logs },
+    components: { nats, logs, livekitClient, peersRegistry },
     request
   } = ctx
 
+  const peerRegistryHandlerByEvent: Record<
+    ParticipantEvent,
+    IPeersRegistry[keyof Pick<IPeersRegistry, 'onPeerConnected' | 'onPeerDisconnected'>]
+  > = {
+    [WebhookEventName.ParticipantJoined]: peersRegistry.onPeerConnected,
+    [WebhookEventName.ParticipantLeft]: (identity: string) => peersRegistry.onPeerDisconnected(identity)
+  }
+
   const logger = logs.getLogger('livekit-webhook')
 
-  const apiKey = await config.requireString('LIVEKIT_API_KEY')
-  const apiSecret = await config.requireString('LIVEKIT_API_SECRET')
+  const body = await request.text()
+  const authorization = request.headers.get('Authorization') || ''
 
-  try {
-    const body = await request.text()
-    const authorization = request.headers.get('Authorization') || ''
+  if (!authorization) {
```

Comment:
Shouldn't we validate the value in `authorization`?

Reply:
The authorization is provided by the livekit webhook, and the `receive` method of the `WebhookReceiver` from the livekit SDK validates the authorization token.